comms/dht/src/dedup/mod.rs

```diff
@@ -79,7 +85,10 @@ where
                 message.tag,
                 message.dht_header.message_tag
             );
-            if dht_requester.insert_message_hash(hash).await? {
+            if dht_requester
+                .insert_message_hash(hash, message.source_peer.public_key.clone())
```

Comment:
Just a note, the source_peer is the source of this message. That is, the peer that it was directly received from perhaps via propagation, but not necessarily the peer that created the original message. You would expect this stored value for the same message to be different across peers. The origin_peer is the authenticated origin (the peer that created the message), but is only available to the peer that can decrypt the message. So I'm not too convinced how useful this field will end up being. Just FYI though, nothing against doing it anyway.

Reply:
Noted